pkg/controller/ipam/antrea_ipam_controller.go

```diff
@@ -113,20 +127,32 @@ func NewAntreaIPAMController(crdClient versioned.Interface,
 	klog.V(2).InfoS("Subscribing for StatefulSet notifications", "controller", controllerName)
 	statefulSetInformer.Informer().AddEventHandler(
 		cache.ResourceEventHandlerFuncs{
+			AddFunc:    c.enqueueStatefulSetCreateEvent,
 			DeleteFunc: c.enqueueStatefulSetDeleteEvent,
 		},
 	)
 
 	return c
 }
 
+// Enqueue the StatefulSet create notification to be processed by the worker
+func (c *AntreaIPAMController) enqueueStatefulSetCreateEvent(obj interface{}) {
+	ss := obj.(*appsv1.StatefulSet)
+	klog.V(2).InfoS("Create notification", "Namespace", ss.Namespace, "StatefulSet", ss.Name)
+
+	key := k8s.NamespacedName(ss.Namespace, ss.Name)
+
+	c.statefulSetMap.Store(key, ss)
+	c.statefulSetQueue.Add(addEventIndication + key)
```

Comment:
Question: any reason for preferring using the first character a string to specify the action to take rather than a struct with a separate field?

Reply:
I think for a single bit of information, serializing and de-serializing struct would be an overkill.

Reply:
yeah, make sense. shall I be very pendant and say that it's not a bit, but a character? :)

Reply:
Why we could not judge ADD/DEL based on the key is in the cache or not, as other controllers do? @tnqn

Reply:
This seemed risky to me: in case object is re-created (delete+create), object may be in store once del event is processed.

Reply:
Humm.. But even in your way if the processing fails and retries, we still lose the state.

Another problem is you delete and store ss to statefulSetMap in different threads, and so it is possible a re-created ss can be deleted mistakenly from the map.

Reply:
I feel we should store/look up previous allocation state in some way, and decide what to do based on the latest ss spec and the allocation state. Then recreation can be handled too.